glyphs-reader/src/font.rs

```diff
@@ -550,20 +590,45 @@ impl RawFont {
         Ok(())
     }
 
-    fn v2_to_v3_shapes(&mut self) -> Result<(), Error> {
+    fn v3_shapes_to_v2_paths_and_components(&mut self) -> Result<(), Error> {
         // shapes in v3 encompasses paths and components in v2
+        // paths and components is closer to what IR wants
+        // so here, unusually, downgrade
         for glyph in self.glyphs.iter_mut() {
             // Paths and components combine in shapes
-            // TODO components
             for layer in glyph.layers.iter_mut() {
-                if let Some(paths) = layer.paths.take() {
-                    match &mut layer.shapes {
-                        Some(shapes) => {
-                            shapes.extend(paths);
-                        }
-                        None => {
-                            layer.shapes = Some(paths);
-                        }
+                let Some(mut shapes) = layer.shapes.take() else {
+                    continue;
+                };
+
+                for shape in shapes.iter_mut() {
+                    if let Some(Plist::String(ref_glyph_name)) = shape.other_stuff.remove("ref") {
+                        // only components use ref
+                        let transform = if let Some(pos) = &shape.pos {
```

Comment:
maybe add a TODO to later on also handle scale and rotation

Reply:
added a general todo for more attributes as there are quite a few. it wasn't immediately obvious which ones we'd need. Certainly scale and rotate :)